net-print/cnijfilter-drivers/cnijfilter-drivers-3.80.ebuild

```diff
@@ -0,0 +1,167 @@
+# Copyright 1999-2012 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+MY_PN="${PN/-drivers/}"
+
+inherit eutils autotools flag-o-matic multilib
+
+DESCRIPTION="Canon InkJet Printer Driver for Linux (Pixus/Pixma-Series)."
+HOMEPAGE="http://support-sg.canon-asia.com/contents/SG/EN/0100469302.html"
+SRC_URI="http://gdlp01.c-wss.com/gds/3/0100004693/01/${MY_PN}-source-${PV}-1.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="${PV}"
+KEYWORDS="~amd64 ~x86"
+PRINTER_USE=( mp230 mg2200 e510 mg3200 mg4200 ip7200 mg5400 mg6300 )
+PRINTER_ID=( 401 402 403 404 405 406 407 408 )
+IUSE="${PRINTER_USE[@]} +servicetools"
+
+RDEPEND="
+	net-print/cnijfilter[servicetools?]
+	>=media-libs/libpng-1.5
+	>=media-libs/tiff-3.4
+	>=net-print/cups-1.4
+	servicetools? (
+		>=dev-libs/libxml2-2.7.3-r2
+		>=x11-libs/gtk+-2.6:2
+	)
+"
+DEPEND="${DEPEND}
+	sys-devel/gettext
+"
+
+REQUIRED_USE="|| ( ${PRINTER_USE[@]} )"
+
+S="${WORKDIR}/${MY_PN}-source-${PV}-1"
+
+_dir_build() {
+	local dirs=$1
+	local command=$2
+	local d
+
+	[[ $# -ne 2 ]] && die "Call as: _dir_build DIRS COMMAND"
+
+	for d in ${dirs}; do
+		local suffix=""
+		echo ">>> Working in: ${d}"
+		pushd ${d} >/dev/null
+		# progpath must be set otherwise we go for /usr/local/bin
+		if [[ ${command} == econf* ]]; then
+			${command} --enable-progpath="${EPREFIX}/usr/bin"
+		else
+			${command}
+		fi
+		popd > /dev/null
+	done
+}
+
+_printer_dir_build() {
+	local command=$1
+	local d
+
+	[[ $# -ne 1 ]] && die "Call as: _printer_dir_build COMMAND"
+
+	for (( i=0; i<${#PRINTER_USE[@]}; i++ )); do
+		local name="${PRINTER_USE[$i]}"
+		if use ${name}; then
+			for d in ${DIRS_PRINTER}; do
+				echo ">>> Working in: ${name}/${d}"
+				pushd ${name}/${d} > /dev/null
+				if [[ ${command} == econf* ]]; then
+					${command} \
+						--enable-progpath="${EPREFIX}/usr/bin" \
+						--program-suffix=${name}
+				else
+					${command}
+				fi
+				popd > /dev/null
+			done
+		fi
+	done
+}
+
+pkg_setup() {
+	[[ -z ${LINGUAS} ]] && LINGUAS="en"
+
+	DIRS_PRINTER="cnijfilter"
+	# lgmon must be first as it is required by cngpijmon
+	use servicetools && DIRS_PRINTER+=" lgmon cngpijmon maintenance"
+}
+
+src_prepare() {
+	local d i
+
+	# missing macros directory make aclocal fail
+	mkdir maintenance/m4 || die
+
+	epatch \
+		"${FILESDIR}/${MY_PN}"-3.70-png.patch \
+		"${FILESDIR}/${MY_PN}"-3.70-ppd.patch \
+		"${FILESDIR}/${MY_PN}"-3.70-ppd2.patch \
+		"${FILESDIR}/${MY_PN}"-3.70-libexec-cups.patch \
+		"${FILESDIR}/${MY_PN}"-3.70-libexec-backend.patch \
+		"${FILESDIR}/${MY_PN}"-3.80-cups.patch
+
+	_dir_build "${DIRS_PRINTER}" "eautoreconf"
+
+	for (( i=0; i<${#PRINTER_USE[@]}; i++ )); do
+		local name="${PRINTER_USE[$i]}"
+		local pid="${PRINTER_ID[$i]}"
```

Comment:
This could be a bit cleaner if you used a single list with values separated by some char instead of two, e.g. something like:

    PRINTERS=( model:id model:id model:id )
    IUSE="${PRINTERS[@]%:*} ..."

    ...
    for p in "${PRINTERS[@]}"; do
      local name=${p%:*}
      local pid=${p#*:}

Most notably, this makes it easier to figure out which is which, without having to match indexes.

Reply:
Fixed